src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/PerformanceTrackerService.cs

```diff
@@ -118,331 +120,30 @@ private bool AllowTelemetry(string analyzerId)
         private void OnSnapshotAdded()
             => SnapshotAdded?.Invoke(this, EventArgs.Empty);
 
-        private static string SnapshotLogger(IEnumerable<AnalyzerPerformanceInfo> snapshots, int unitCount)
+        private static string SnapshotLogger(IEnumerable<AnalyzerPerformanceInfo> snapshots, int unitCount, bool forSpan)
         {
             using var pooledObject = SharedPools.Default<StringBuilder>().GetPooledObject();
             var sb = pooledObject.Object;
 
             sb.Append(unitCount);
 
+            sb.Append('(');
+            sb.Append(forSpan ? "SpanAnalysis" : "DocumentAnalysis");
+            sb.Append(')');
+
             foreach (var snapshot in snapshots)
             {
-                sb.Append("|");
+                sb.Append('|');
                 sb.Append(snapshot.AnalyzerId);
-                sb.Append(":");
+                sb.Append(':');
                 sb.Append(snapshot.BuiltIn);
-                sb.Append(":");
+                sb.Append(':');
                 sb.Append(snapshot.TimeSpan.TotalMilliseconds);
             }
 
-            sb.Append("*");
+            sb.Append('*');
 
             return sb.ToString();
         }
-
-        private sealed class ReportGenerator : IDisposable, IComparer<ExpensiveAnalyzerInfo>
```

Comment:
This is the expensive computation part for finding expensive analyzers, which was extremely expensive and slow. This has now been deleted and instead we report aggregated average/stddev for the last set of samples without any expensive computation.